contracts/01-tamagotchi/tests/smoke.rs

```diff
@@ -1,10 +1,39 @@
-use gtest::{Program, System};
+use gtest::{Log, Program, System};
+use tamagotchi_io::*;
 
 #[test]
 fn smoke_test() {
     let sys = System::new();
     sys.init_logger();
-    let _program = Program::current(&sys);
+    let program = Program::current(&sys);
+    let result = program.send(2, String::from("Goodmoring"));
+    assert!(!result.main_failed());
+    let result = program.send(2, TmgAction::Name);
+    let log = Log::builder()
+        .dest(2)
+        .payload(TmgEvent::Name(String::from("Goodmoring")));
+    assert!(result.contains(&log));
+    let _result = program.send(2, TmgAction::Age);
+    // let log = Log::builder().dest(2).payload(TmgEvent::Age(sys.block_timestamp()));
+    // assert!(result.contains(&log));
 
-    // TODO: 8️⃣ Test the program initialization and message handling
+    //How to test the age?
+}
+
+#[test]
+fn negative_smoke_test() {
+    let sys = System::new();
+    sys.init_logger();
+    let program = Program::current(&sys);
+    let payload = vec![1, 2, 3];
+    let _result = program.send(2, payload);
+    // assert!(result.main_failed());
+    // Why the assert is panic?
```

Comment:
It panics because the `init()` function can't decode the payload as a UTF-8 string, I guess.